File: packages/adapters/src/message.ts
export type Message = {
  /** Stable, platform-native — needed for dedup and deep links. */
  id: string;
  conversationId: string;
  author: {
    id: string;
    displayName: string;
  };
  /** Plaintext; keep rich content out of v1. */
  text: string;
  timestamp: Date;
  /** Non-negotiable — trust depends on it. */
  permalink: string;
};

File: services/extraction/src/message.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Author {
    pub id: String,
    pub display_name: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Message {
    pub id: String,
    pub conversation_id: String,
    pub author: Author,
    pub text: String,
    pub timestamp: String,
    pub permalink: String,
}
